src/Icicle/Avalanche/Check.hs

```diff
@@ -88,7 +88,13 @@ checkStatement frag ctx stmt
                    $ checkExp frag (ctxExp ctx) x
                Let n x' <$> go stmts
 
-        ForeachInts n from to stmts
+        While t n to stmts
+         -> do to'   <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) to
+
+               While t n to' <$> go stmts
```

Comment:
ok. so if the way you plan on using this is:
```
Init n = initx
While n to
  ...
  Write n updx
```
this should look up `n` in the accumulator environment, and check that the while expression `to` has the same type as `n`. It doesn't need to add `n` to the expression environment below